### PR TITLE
Started Configurations Revamp in new branch 

### PR DESCRIPTION
**Pull request by angelac2023**
We would really appreciate some feedback on our progress.

### CODE REVIEW

File: leaf-analysis/app.js
Comment (leshghi) [inline leaf-analysis/app.js]: Please remove this file from PR

File: leaf-ui/css/style.css
Comment (leshghi) [inline leaf-ui/css/style.css]: Why do we have empty fields in the css?

File: leaf-ui/js/onFunctions.js
Comment (leshghi) [inline leaf-ui/js/onFunctions.js]: I would remove the console logs and add a TODO to implement

**Review by leshghi — COMMENTED** [review]
This looks like a good start to me. The biggest issue for me currently is that when you select a config and then switch back the params resets back to the default values, and you need to click again to see the config's actual values. Please also remove the now unused functions in AnalysisInspector.js. There are also some CSS things I personally would change (big gap between analysis params title and input boxes, maybe a line between the analysis params area and the config sidebar, input boxes too narrow, etc) but those are minor things and can be dealt with later. I defer to @amgrubb's review on what they want to change/prioritize.

File: leaf-analysis/app.js
Comment (amgrubb) [reply leaf-analysis/app.js]: Agreed this should be removed.

**Review by amgrubb — CHANGES_REQUESTED** [review]
Please make these updates, so I can run it locally.

File: leaf-analysis/app.js
Comment (leshghi) [inline leaf-analysis/app.js]: Remove this file from commit

File: leaf-ui/css/style.css
Comment (leshghi) [reply leaf-ui/css/style.css]: ^ @angelac2023 @Manarnz

File: leaf-ui/js/onFunctions.js
Comment (leshghi) [inline leaf-ui/js/onFunctions.js]: Remove this

File: leaf-ui/js/onFunctions.js
Comment (leshghi) [reply leaf-ui/js/onFunctions.js]: ^ @angelac2023 , @Manarnz

File: leaf-ui/rappid-extensions/BackboneModelsAnalysis.js
Comment (leshghi) [inline leaf-ui/rappid-extensions/BackboneModelsAnalysis.js]: What does mode represent?

File: leaf-ui/rappid-extensions/ConfigInspector.js
Comment (leshghi) [inline leaf-ui/rappid-extensions/ConfigInspector.js]: I don't know if these comments are necessary/accurate. I defer to @amgrubb on that though. If you leave them in the correct style is `// First word capitalized`

File: leaf-ui/rappid-extensions/ConfigInspector.js
Comment (leshghi) [inline leaf-ui/rappid-extensions/ConfigInspector.js]: Same comment as above re: these comments

File: leaf-ui/rappid-extensions/ConfigInspector.js
Comment (leshghi) [inline leaf-ui/rappid-extensions/ConfigInspector.js]: Remove

**Review by leshghi — CHANGES_REQUESTED** [review]
I like the concept - for me the biggest bug is that now when you delete the first config the entire sidebar disappears when you switch to modelling mode and then back to analysis mode. Here is a thought - you already have a selected parameter - what if you just used that instead of mode to make sure the previously selected configuration is the one that has its analysisInspector rendered?

File: leaf-analysis/app.js
Comment (leshghi) [inline leaf-analysis/app.js]: Can you please remove this file from your commit, thanks

Comment (leshghi) [reply leaf-analysis/app.js]: Also deleting this line makes it impossible to actually run (and therefore test) the web page.

File: leaf-ui/css/style.css
Comment (leshghi) [inline leaf-ui/css/style.css:150]: Why did you remove the hovering feature and would it be possible to bring it back?

File: leaf-ui/index.html
Comment (leshghi) [inline leaf-ui/index.html:139]: Remove this line

File: leaf-ui/js/onFunctions.js
Comment (leshghi) [inline leaf-ui/js/onFunctions.js]: Remember my previous review where I mentioned correct comment format is // First word capitalized

File: leaf-ui/js/onFunctions.js
Comment (leshghi) [inline leaf-ui/js/onFunctions.js]: Same thing about commenting convention

File: leaf-ui/js/onFunctions.js
Comment (leshghi) [inline leaf-ui/js/onFunctions.js]: Why did you move this commented out code?

File: leaf-ui/js/onFunctions.js
Comment (leshghi) [inline leaf-ui/js/onFunctions.js]: No need to touch these lines

File: leaf-ui/js/onFunctions.js
Comment (leshghi) [inline leaf-ui/js/onFunctions.js]: No need to touch this line

File: leaf-ui/js/onFunctions.js
Comment (leshghi) [inline leaf-ui/js/onFunctions.js]: Same thing about commenting convention

File: leaf-ui/js/onFunctions.js
Comment (leshghi) [inline leaf-ui/js/onFunctions.js]: If we are re-sectioning these functions into appear and disappear, why is this in a separate section?

**Review by leshghi — COMMENTED** [review]
Once you respond to the comments and make the changes I will approve. Also, #465 should follow this PR if you are looking for something to work on once it is merged.

File: leaf-ui/css/style.css
Comment (angelac2023) [reply leaf-ui/css/style.css:150]: We initially removed it because it was making the configurations sidebar look rather odd. We will add it back in the next commit.

**Review by leshghi — APPROVED** [review]
Thanks for making these changes

File: leaf-ui/js/onFunctions.js
Comment (leshghi) [inline leaf-ui/js/onFunctions.js]: Remove this line

File: leaf-ui/rappid-extensions/AnalysisInspector.js
Comment (amgrubb) [inline leaf-ui/rappid-extensions/AnalysisInspector.js:87]: Note: This code should not have been deleted.

**Review by amgrubb — APPROVED** [review]

**Pull request merged**
